src/component/DefaultTooltipContent.tsx

```diff
@@ -50,21 +50,24 @@ export interface Props<TValue extends ValueType, TName extends NameType> {
   itemSorter?: (item: Payload<TValue, TName>) => number | string;
 }
 
-export class DefaultTooltipContent<TValue extends ValueType, TName extends NameType> extends PureComponent<
-  Props<TValue, TName>
-> {
-  static displayName = 'DefaultTooltipContent';
-
-  static defaultProps = {
-    separator: ' : ',
-    contentStyle: {},
-    itemStyle: {},
-    labelStyle: {},
-  };
-
-  renderContent() {
-    const { payload, separator, formatter, itemStyle, itemSorter } = this.props;
+export const DefaultTooltipContent = <TValue extends ValueType, TName extends NameType>(
+  props: Props<TValue, TName>,
+) => {
+  const {
+    separator = ' : ',
+    contentStyle = {},
+    itemStyle = {},
+    labelStyle = {},
```

Comment:
Just a note, anything that is called within a `cloneElement` within `generateCategoricalChart` has the potential to lose the values of their default props (we've discussed this issue elsewhere I believe I forget where atm though). I think we're good since this is within the function though https://github.com/recharts/recharts/blob/master/src/chart/generateCategoricalChart.tsx#L1929-L1946

Reply:
Yeah I believe the issue was with `defaultProps` property on the component, but since these are in the function are fine.

Its a bit hard to apply this style of defaultProps since the `props` object has been passed around a lot in all components, its better to make the functions only receive the props they need in new refactors (I'd do them as I got time if no one picked it up).